File: php/src/Exceptions/RateLimitException.php
<?php

declare(strict_types=1);

namespace TempMail\Exceptions;

class RateLimitException extends TempMailException
{
    public function __construct(
        string $message = 'Rate limit exceeded',
        public readonly ?int $retryAfter = null,
        array $context = [],
    ) {
        parent::__construct($message, 429, context: $context);
    }
}

File: java/src/main/java/com/tempmail/errors/RateLimitException.java
package com.tempmail.errors;

public class RateLimitException extends TempMailException {

    private final int retryAfterSeconds;

    public RateLimitException(String message) {
        super(message, 429);
        this.retryAfterSeconds = -1;
    }

    public RateLimitException(String message, int retryAfterSeconds) {
        super(message, 429);
        this.retryAfterSeconds = retryAfterSeconds;
    }

    public RateLimitException(String message, Throwable cause) {
        super(message, 429, cause);
        this.retryAfterSeconds = -1;
    }

    public int getRetryAfterSeconds() {
        return retryAfterSeconds;
    }
}
